### ISSUE TITLE
Missing documented property of StyleClass on VisualElement

### ISSUE THREAD

**Issue by davidglassborow** [fsprojects/Fabulous#205]
The [styles](/fsprojects/Fabulous/blob/master/docs/views-styling.md#css-styling-with-xamarinforms-30) documentation uses a styleClass property that doesn't seem to exist on any of the view objects.

I'm keen to do a PR to add it, so:

- The underlying property is on [VisualElement](https://docs.microsoft.com/en-gb/dotnet/api/xamarin.forms.visualelement.styleclass?view=xamarin-forms) so I'm guessing we need a new entry in Xamarin.Forms.Core.Json under `Xamarin.Forms.VisualElement`
- The underlying control is an `IList<string>` but the docs suggest a simple string, so should I be splitting the F# string or similar when applying to the underlying control ?
- How do I run the generator ?  It doesn't seem to run as any of the fake targets, and running from the CLI or VS Mac leads to a `Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'FSharp.Core, Version=4.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.` error

**Comment by TimLariviere** [comment]
Hi @davidglassborow 

Thanks for your interest in Fabulous. :)
`styleClass` is indeed missing.

Based on your remarks, I've fixed a few things with the generator. (see https://github.com/fsprojects/Fabulous/pull/206 - pending merge)
- No more missing FSharp.Core
- FAKE target to run the generator (docs here : https://github.com/fsprojects/Fabulous/pull/206/commits/b868fe8fd0118a68ac3bee05ef26d60d66ac7f89)

>The underlying property is on VisualElement so I'm guessing we need a new entry in Xamarin.Forms.Core.Json under Xamarin.Forms.VisualElement

Yes, exactly.

>The underlying control is an IList<string> but the docs suggest a simple string, so should I be splitting the F# string or similar when applying to the underlying control ?

I think splitting the string would not be the best option.
It would be better to use a list of string, but what the docs of Fabulous suggest is also a valid option (easier to write).
eg.
```fsharp
View.Label(text = "Hello", styleClass = "myStyle")
View.Label(text = "Hello", styleClass = [  "myStyleA"; "myStyleB" ])
```

You can take a look at how Margin is implemented (supports both string and Thickness)
https://github.com/fsprojects/Fabulous/blob/df42a6dfd6f5e94cab20daf0bc1ff0319cb159b8/tools/Generator/Xamarin.Forms.Core.json#L148-L152

Basically, `inputType` would be `string list`, `modelType` would be `string[]`, and `convToModel` would be a new converter function with a pattern matching for string and string list.

For example, you will find `makeThickness` in ViewConverters.fs inside the Fabulous.Core project.
https://github.com/fsprojects/Fabulous/blob/df42a6dfd6f5e94cab20daf0bc1ff0319cb159b8/src/Fabulous.Core/ViewConverters.fs#L158-L162

Unfortunately based on what I saw, the Generator will create invalid code for `string list`.
It assumes every list should be of type `ViewElement list`.
So the generator need to be updated before to support that scenario.
https://github.com/fsprojects/Fabulous/blob/df42a6dfd6f5e94cab20daf0bc1ff0319cb159b8/tools/Generator/Program.fs#L434-L435

**Comment by davidglassborow** [comment]
Thanks @TimLariviere that's very helpful.

I've raised PR with my initial attempt at getting this working, including an example styled label added to the about page on AllControls.

Hopefully this is good to merge, shout if you think my approach is wrong.

**Comment by dsyme** [comment]
fixed by #209
